File: src/main/java/com/gitlab/lae/intellij/actions/tree/ActionList.java
package com.gitlab.lae.intellij.actions.tree;

import com.intellij.ui.components.JBList;

import java.awt.event.KeyEvent;
import java.util.Collection;

final class ActionList<E> extends JBList<E> {

    ActionList(Collection<E> items) {
        super(items);
    }

    @Override
    protected void processKeyEvent(KeyEvent e) {
        e.setSource(this);
        super.processKeyEvent(e);
    }

}

File: src/main/kotlin/com/gitlab/lae/intellij/actions/tree/ActionList.kt
package com.gitlab.lae.intellij.actions.tree

import com.intellij.ui.components.JBList
import java.awt.event.KeyEvent

class ActionList<E>(items: Collection<E>) : JBList<E>(items) {

    public override fun processKeyEvent(e: KeyEvent) {
        e.source = this
        super.processKeyEvent(e)
    }
}
